polaris-core/src/main/java/org/apache/polaris/core/storage/cache/StorageCredentialCache.java

```diff
@@ -50,13 +50,10 @@ public class StorageCredentialCache {
   private static final long CACHE_MAX_NUMBER_OF_ENTRIES = 10_000L;
 
   private final LoadingCache<StorageCredentialCacheKey, StorageCredentialCacheEntry> cache;
-  private final RealmContext realmContext;
   private final PolarisConfigurationStore configurationStore;
 
   /** Initialize the creds cache */
-  public StorageCredentialCache(
-      RealmContext realmContext, PolarisConfigurationStore configurationStore) {
-    this.realmContext = realmContext;
+  public StorageCredentialCache(PolarisConfigurationStore configurationStore) {
     this.configurationStore = configurationStore;
     cache =
         Caffeine.newBuilder()
```

Comment:
Would you mind making the max cache size (by entries) configurable? Now that the caches is shared across realms, configuring its side is more important to users, I guess.

I suppose the cache size has to be a global configuration (maybe a constructor parameter injected via CDI producers + Quarkus config).

Reply:
i've added a commit for this, is this what you had in mind?
please double check the naming, module and package.

also imo this raises some questions:

a) how should the new cache config relate to `FeatureConfiguration.STORAGE_CREDENTIAL_CACHE_DURATION_SECONDS` and `FeatureConfiguration.STORAGE_CREDENTIAL_DURATION_SECONDS` ?

b) what to do with the `InMemoryEntityCache` which seems to deal with a single realm? (and can thus its max size cant be configured at the application level)

Reply:
> InMemoryEntityCache

I was thinking about that one, too. And I suspect it should be refactored in a similar way.

Reply:
InMemoryEntityCache -> +1 to similar refactoring (better control over heap usage)